scripts/irods/test/test_irsync.py

```diff
@@ -805,3 +805,37 @@ def test_irsync_with_all_keyword__issue_8613(self):
         finally:
             self.user0.run_icommand(['irm', '-f', data_object_path])
             self.admin.run_icommand(['iadmin', 'rmresc', another_resc])
+
+    def test_irsync_overwrites_stale_replicas__issue_8590(self):
+        source_coll = 'thesource_8590'
+        dest_coll = 'thetarget_8590'
+        data_object_name = 'stale_replica_file_8590'
+
+        test_file_path = os.path.join(self.testing_tmp_dir, data_object_name)
+
+        source_coll_path = f'{self.user0.session_collection}/{source_coll}'
+        dest_coll_path = f'{self.user0.session_collection}/{dest_coll}'
+
+        dest_data_obj_path = f'{dest_coll_path}/{data_object_name}'
+
+        with open(test_file_path, 'w') as f:
+            f.write('potato')
+
+        self.user0.assert_icommand(['imkdir', source_coll_path])
+        self.user0.assert_icommand(['imkdir', dest_coll_path])
+
+        self.user0.assert_icommand(['iput', test_file_path, f'{source_coll_path}/{data_object_name}'])
+        self.user0.assert_icommand(['itouch', f'{dest_coll_path}/{data_object_name}'])
+
+        self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', dest_data_obj_path, 'replica_number', '0', 'DATA_REPL_STATUS', '0'])
+
+        # Should not fail with SYS_NO_GOOD_REPLICA
+        self.user0.assert_icommand(['irsync', '-K', '-v', '-r', f'i:{source_coll_path}', f'i:{dest_coll_path}' ], 'STDOUT_SINGLELINE', data_object_name)
```

Comment:
what SHOULD have happened here?   is the target now complete and good?   please assertTrue that after this happens, if so.